src/components/SwapTokenInput.tsx

```diff
@@ -0,0 +1,289 @@
+import { ArrowDropDown } from "@mui/icons-material"
+import Autocomplete from "@mui/material/Autocomplete"
+import { BigNumber } from "ethers"
+import Box from "@mui/material/Box"
+import ClickAwayListener from "@mui/material/ClickAwayListener"
+import Popper from "@mui/material/Popper"
+import Search from "@mui/icons-material/Search"
+import { TokenOption } from "../pages/Swap"
+import { useTranslation } from "react-i18next"
+
+const StyledPopper = styled(Popper)(({ theme }) => ({
+  border: `1px solid ${theme.palette.other.divider}`,
+  borderRadius: theme.shape.borderRadius,
+  width: 400,
+  padding: theme.spacing(2),
+  zIndex: theme.zIndex.modal,
+  backgroundColor: theme.palette.background.paper,
+}))
+
+interface SwapTokenInputProps {
+  tokens: TokenOption[]
+  selected?: string
+  inputValue: string
+  inputValueUSD: BigNumber
+  isSwapFrom?: boolean
+  onSelect?: (tokenSymbol: string) => void
+  onChangeAmount?: (value: string) => void
+}
+
+export default function SwapTokenInput({
+  tokens,
+  selected,
+  inputValue,
+  inputValueUSD,
+  onSelect,
+  onChangeAmount,
+}: SwapTokenInputProps): ReactElement {
+  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null)
+  const containerRef = useRef<HTMLDivElement>(null)
+  const theme = useTheme()
+
+  const handleClick = () => {
+    // setAnchorEl(event.currentTarget)
+    if (containerRef?.current) setAnchorEl(containerRef.current)
+  }
+
+  const handleClose = () => {
+    if (anchorEl) {
+      anchorEl.focus()
+    }
+    setAnchorEl(null)
+  }
+
+  const handleInputChange = (
+    e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>,
+  ) => {
+    const re = /^[0-9]*[.,]?[0-9]*$/
+    // if value is not blank, then test the regex
+    if (e.target.value === "" || re.test(e.target.value)) {
+      onChangeAmount?.(e.target.value)
+    }
+  }
+
+  const handleFocus = (
+    event: React.FocusEvent<HTMLInputElement | HTMLTextAreaElement, Element>,
+  ) => {
+    event.target.select()
+  }
+
+  const open = Boolean(anchorEl)
+  const selectedToken =
+    typeof selected === "string" ? TOKENS_MAP[selected] : undefined
+
+  return (
+    <div>
+      <Box
+        display="flex"
+        alignItems="center"
+        padding={1}
+        borderRadius="6px"
+        border={`1px solid ${theme.palette.other.border} `}
+        bgcolor={theme.palette.background.paper}
+        ref={containerRef}
+      >
+        {selectedToken?.icon && (
+          <Box width={24} height={24} marginRight={1}>
+            <img
+              src={selectedToken?.icon}
+              alt={selectedToken?.name}
+              width="100%"
+              height="100%"
+            />
+          </Box>
+        )}
+
+        <Box flexWrap="nowrap">
+          <Button
+            disableRipple
+            onClick={handleClick}
+            endIcon={<ArrowDropDown />}
+            data-testid="listOpenBtn"
+            disableElevation
+            disableFocusRipple
```

Comment:
disableElevation and disableFocusRipple should be disabled globally at the theme level, correct?

Reply:
Exactly!